ludwig/features/image_feature.py

```diff
@@ -50,6 +51,67 @@ def get_feature_meta(column, preprocessing_parameters):
             'preprocessing': preprocessing_parameters
         }
 
+    @staticmethod
+    def _read_image_and_resize(filepath,
+                               im_width,
+                               im_height,
+                               should_resize,
+                               num_channels,
+                               resize_method,
+                               user_specified_num_channels):
+        """
+        :param filepath: path to the image
+        :param im_width: expected width of the image
+        :param im_height: expected height of the image
+        :param should_resize: Should the image be resized?
+        :param resize_method: type of resizing method
+        :param num_channels: expected number of channels in the first image
+        :param user_specified_num_channels: did the user specify num channels?
+        :return: image object
+
+        Helper method to read and resize an image according to model defn.
+        If the user doesn't specify a number of channels, we use the first image
+        in the dataset as the source of truth. If any image in the dataset
+        doesn't have the same number of channels as the first image,
+        raise an exception.
+
+        If the user specifies a number of channels, we try to convert all the
+        images to the specifications by dropping channels/padding 0 channels
+        """
+
+        img = imread(filepath)
+        im_num_channels = num_channels_in_image(img)
+        if im_num_channels == 1:
+            img = img.reshape((img.shape[0], img.shape[1], 1))
+
+        if user_specified_num_channels is False:
+            # If the image isn't like the first image, raise exception
+            if im_num_channels != num_channels:
+                raise ValueError(
+                    'Image {0} has {1} channels, unlike the first image, which'
+                    ' has {2} channels'.format(filepath,
+                                               im_num_channels,
+                                               num_channels))
+
+        else:
+            # Number of channels is specified by the user
+            img_padded = np.zeros((im_height, im_width, num_channels))
+            for i in range(min(num_channels, im_num_channels)):
```

Comment:
This for loop is slow, we should compute the `min_num_channels` beforehand and then assign with `img_padded[:, :, :min_num_channels] = img[:, :, :min_num_channels]`

Reply:
makes sense. will update.